marshal/tpmu-types.c

```diff
@@ -308,10 +307,12 @@ TSS2_RC Tss2_MU_##type##_Marshal(type const *src, uint32_t selector, uint8_t buf
     return ret; \
 }
 
-#define TPMU_MARSHAL2(...) TPMU_MARSHAL(__VA_ARGS__, -1, ADDR, na, marshal_null, -2, ADDR, na, marshal_null,\
-                                        -3, ADDR, na, marshal_null, -4, ADDR, na, marshal_null, -5, ADDR, na, marshal_null, \
-                                        -6, ADDR, na, marshal_null, -7, ADDR, na, marshal_null, -8, ADDR, na, marshal_null, \
-                                        -9, ADDR, na, marshal_null, -10, ADDR, na, marshal_null)
+#define TPMU_MARSHAL2(type, sel, op, m, fn, ...) \
+    TPMU_MARSHAL(type, sel, op, m, fn, __VA_ARGS__, -1, ADDR, m, marshal_null, \
```

Comment:
You don't need this op field. You can just move it into "m".

What I did is just expose src/dest in the macros and then you send the member and all through.


Ie:
```
TPMU_MARSHAL(FOO, BAR1, &dest->bar, marshal_bar, BAR2, &dest->bar2, marshal_bar2,...)
```

I think it's simpler and more flexible but is a larger change.